Danki2/Assets/Scripts/Actor/Services/MovementManager.cs

```diff
@@ -0,0 +1,56 @@
+﻿using UnityEngine;
+
+public class MovementManager : MonoBehaviour
+{
+    float _moveLockRemainingDuration;
+    float _moveLockSpeed;
+    bool _moveLockOn;
+    Vector3 _moveLockDirection;
+    Vector3 _moveDirection;
+
+    void Start()
+    {
+        _moveLockRemainingDuration = 0f;
+        _moveLockSpeed = 0f;
+        _moveLockOn = false;
+        _moveLockDirection = Vector3.one;
+        _moveDirection = Vector3.zero;
+    }
+
+    void Update()
+    {
+        _moveLockRemainingDuration = Mathf.Max(0f, _moveLockRemainingDuration - Time.deltaTime);
+        _moveLockOn = _moveLockRemainingDuration > 0 ? _moveLockOn : false;
+    }
+
+    private void LateUpdate()
```

Comment:
Player movement wants to be done in normal Update(), but with Interpolation turned on in the Players Rigidbody component. This will smooth out the jerkyness.

Reply:
The problem is, how can we guarentee the move doesn't happen until the move has been registered each frame? I think we can expose a Move method that is called at the end of Actor.Update - but annoyingly this would expose it to everything else...

Reply:
Because each entity that moves will just decide what movement it wants then call to execute the movement.
In the players case, it's update will be like: 
HandleInput();
Move();

Reply:
Done all your other comments, still not convinced I agree with this one.

This is something that every actor will want to do at the end of each Update, so it belongs there for me.